File: math/src/wall.cpp
#include "wall.h"

Wall::Wall()
{
	(void)0;
}
Wall::Wall(sf::Vector2f p1, sf::Vector2f p2) 
{
	this->p1.x = p1.x;
	this->p2.x = p2.x;
	this->p1.y = p1.y;
	this->p2.y = p2.y;  
	this->line = sf::VertexArray(sf::Lines, 2);
    this->line[0].position = this->p1;
    this->line[1].position = this->p2;
    this->calculateEquation();
}

double Wall::getSlope() 
{
   return this->slope;
};

double Wall::getIntercept() 
{
   return this->intercept;
};

void Wall::calculateEquation() {
    // Check if the line is vertical
    if (x1 != x2) {
        // Calculate the slope (m) and y-intercept (b)
        slope = (y2 - y1) / (x2 - x1);
        intercept = y1 - slope * x1;
    } else {
        // If the line is vertical, store slope as NaN and x-intercept
        slope = NAN;
        intercept = x1;
    }
};

void Wall::printEquation() {
    if (!std::isnan(slope)) {
        std::cout << "y = " << slope << "x + " << intercept << std::endl;
    } else {
        std::cout << "x = " << intercept << std::endl;
    }
};

void Wall::draw(sf::RenderWindow *window)
{
	window->draw(this->line);
}

File: python/src/wall.py
import pygame
import math

class Wall:
    def __init__(self, cords):
        self.pos = cords
        self.slope = 0
        self.intercept = 0
        self.calculateEquation()

    def draw(self, screen):
        pygame.draw.line(screen, (255,255,255), self.pos[0], self.pos[1], 5)

    def calculateEquation(self): 
        if self.pos[0].x != self.pos[1].x:
            self.slope = (self.pos[1].y - self.pos[0].y) / (self.pos[1].x - self.pos[0].x)
            self.intercept = self.pos[0].y - self.slope * self.pos[0].x
        else:
            self.slope = None
            self.intercept = self.pos[0].x

    def printEquation(self):
        if self.slope != None:

            print("y =", self.slope, "x +", self.intercept)
        else:
            print("x =", self.intercept)